src/util/DataResolver.js

```diff
@@ -34,7 +34,7 @@ class DataResolver extends null {
    * @returns {string}
    */
   static resolveCode(data, regex) {
-    return data.matchAll(regex).next().value?.[1] ?? data;
+    return new RegExp(regex.source).exec(data)?.[1] ?? data;
```

Comment:
Ideally, we should check for [`RegExp.prototype.global`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global) before creating a clone.